src/dso_api/dynamic_api/serializers/base.py

```diff
@@ -234,21 +214,39 @@ def _request(self):
 
     def get_fields(self) -> dict[str, serializers.Field]:
         """Override DRF to remove fields that shouldn't be part of the response."""
-        fields = {}
-        for field_name, field in super().get_fields().items():
-            if self._apply_permission(field_name, field):
-                fields[field_name] = field
-
-        return fields
+        base_fields = super().get_fields()
+        return {
+            field_name: field
+            for field_name, field in base_fields.items()
+            if field_name in self.fields_always_included
+            or self._apply_permission(field_name, field)
+        }
 
     def _apply_permission(self, field_name: str, field: serializers.Field) -> bool:
         """Check permissions and apply any transforms."""
         user_scopes = self._request.user_scopes
 
-        if field.source == "*":
-            # e.g. _links field, always include. These sub serializers
-            # do their own permission checks for their fields.
+        if field.source == "*" and isinstance(field, serializers.Serializer):
+            # e.g. _links field or "schema" field, always include.
+            # The sub serializers do their own permission checks for their fields.
+            # Anything else (even with source="*") passes through to enforce checks.
             return True
+        elif isinstance(field, FieldAccessMixin) and not user_scopes.has_table_access(
+            field.Meta.model.table_schema()
+        ):
+            # Extra case for the _links section: when a LinkSerializer object will not render
+            # any fields (due to not having access to the table), don't render that block at all.
+            # Otherwise, it would just render `"relation": {}`. Now the field is simply missing,
+            # in the same way other body fields are omitted when there is no access.
+            #
+            # When a user has access to a relation that is NULL, it will still be rendered
+            # as "relation": null, or "relation": [] for M2M, so there is no ambiguity here.
+            logging.debug(
```

Comment:
```suggestion
            logging.info(
```

Again, security-related.